src/App.tsx

```diff
@@ -1,34 +1,45 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions } from './features/todos';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const dispatch = useAppDispatch();
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(todosFromServer => {
+        dispatch(actions.setTodos(todosFromServer));
+      })
+      .finally(() => setLoading(false));
+  }, []);
```

Comment:
```suggestion
  }, [dispatch]);
```
Do not ignore it) Dispatch is memorized, so your useEffect callback will also run only once.
![image](https://github.com/mate-academy/react_redux-list-of-todos/assets/86923089/db8455c9-1adf-44d4-9468-97e656219c18)